Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.h

```diff
@@ -28,7 +31,10 @@ namespace ROS2
     public:
         JointsManipulationComponent();
         JointsManipulationComponent(
-            const PublisherConfiguration& configuration, const AZStd::unordered_map<AZStd::string, JointPosition>& initialPositions);
+            const PublisherConfiguration& configuration,
+            const AZStd::unordered_map<AZStd::string, JointPosition>& initialPositions,
+            AZStd::vector<AZStd::string> jointNames,
```

Comment:
Why is redundant joint names ? We have joint names in the map.

Reply:
We have joint names in the map, but: 
- they are keys of hashmap, and ordering is important for position control, as the message used is just array of floats

- the order in which the map is filled is BFS, but order of siblings is not specified (Bus::Event is used, where the policy doesn't specify order of handlers). Because of that, if you have eg. multiple gripper joints, they will change they position as you add/remove their siblings.


This was more natural to me, as in controllers.yaml file user would specify the joints as a list.
(This will also be needed if in the future we will want to have separate topics for controlling parts of the robot - e.g. separate controllers for arm/gripper).

If your suggestion is to use the map instead, I can work through the changes necessary to make the ordering more predictable.

Reply:
What would these changes be for a map? Alphabetical order of joints might not be desired, perhaps they would typically be indexed from top-level down the kinematic chain?

If alphabetical order is bad, I think that the vector can stay, or that we could map joints to position&index. Perhaps it would be better rename the vector to indicate it is about indexes (m_orderedJointNames or sth)?

Whatever the structure is, we should populate it instead of requiring user manual input, especially when the information that user needs to input is redundant.